File: 프로그래머스/기초 트레이닝 - Level 0/수열과 구간 쿼리2-solve.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    private static final int INF = 1000001;
    
    public int[] solution(int[] arr, int[][] queries) {
        List<Integer> answer = new ArrayList<>(); 
        for(int[] query : queries) {
            int minNum = INF; 
            for (int i = query[0]; i <= query[1]; i++) {
                if (arr[i] > query[2]) minNum = Math.min(minNum, arr[i]); 
            }
            answer.add(minNum != INF ? minNum : -1);
        }
        
        return answer.stream().mapToInt(i -> i).toArray(); 
    }
}

File: 프로그래머스/기초 트레이닝 - Level 0/수열과 구간 쿼리2-solve.py
def solution(arr, queries):
    INF = 1000001
    answer = []
    for s, e, k in queries: 
        min_num = INF
        for i in range(s, e + 1): 
            if arr[i] > k:
                min_num = min(min_num, arr[i])
        answer.append(min_num if min_num != INF else -1)
    return answer